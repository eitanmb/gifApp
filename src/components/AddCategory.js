import React from 'react';
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleValue = (e) => {
        setInputValue( e.target.value );

        //para fines de las pruebas
        console.log('handleValue', inputValue);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        console.log('handleSubmit', inputValue);

        if(inputValue.trim().length > 2 ) {
            setCategories ( catsExistentes => [ inputValue, ...catsExistentes] );
            setInputValue('');
        }
    }
    
    return (
        <form onSubmit={ e => handleSubmit(e) }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                onChange= { (e) => handleValue(e) }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
