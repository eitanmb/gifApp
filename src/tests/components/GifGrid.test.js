import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en <GifGrid />', () => {
    
    const category = 'One Punch';

    test('debe hacer bien el render', () => {
    
        useFetchGifs.mockReturnValue( {
            data:[],
            loading: true
        });
       
        const wrapper = shallow( <GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe mostrar un item luego de ejecutar useFetchGifs', () => {

        const gifs = [
            {
                id: "ABC",
                title: "Mi Imagen",
                url: "https://localhos/miimagen.jpg"
             },
    
            {
                id: "FGFG",
                title: "Mi Imagen",
                url: "https://localhos/miimagen.jpg"
            }
        ] ;

        useFetchGifs.mockReturnValue( {
            data:gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    
    
    
})
