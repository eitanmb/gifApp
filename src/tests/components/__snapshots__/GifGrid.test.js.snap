// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe hacer bien el render 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar un item luego de ejecutar useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Mi Imagen"
      url="https://localhos/miimagen.jpg"
    />
    <GifGridItem
      id="FGFG"
      key="FGFG"
      title="Mi Imagen"
      url="https://localhos/miimagen.jpg"
    />
  </div>
</Fragment>
`;
